skrub/_agg_joiner.py

```diff
@@ -0,0 +1,597 @@
+
+    suffix : str or iterable of str, default=None
+        The suffixes that will be add to each table columns in case of
+        duplicate column names, similar to `("_x", "_y")` in :obj:`~pandas.merge`.
+        If set to None, we will use the table index, by looking at their
+        order in 'tables', e.g. for a duplicate columns: price (main table),
+        price_1 (auxiliary table 1), price_2 (auxiliary table 2), etc.
+
+    See Also
+    --------
+    AggTarget :
+        Aggregates the target `y` before joining its aggregation
+        on the base dataframe.
+
+    Joiner :
+        Augments a main table by automatically joining multiple
+        auxiliary tables on it.
+
+    Examples
+    --------
+    >>> main = pd.DataFrame({
+            "airportId": [1, 2],
+            "airportName": ["Paris CDG", "NY JFK"],
+        })
+    >>> aux = pd.DataFrame({
+            "flightId": range(1, 7),
+            "from_airport": [1, 1, 1, 2, 2, 2],
+            "total_passengers": [90, 120, 100, 70, 80, 90],
+            "company": ["DL", "AF", "AF", "DL", "DL", "TR"],
+        })
+    >>> join_agg = AggJoiner(
+            table=aux,
+            foreign_key="from_airport",
+            main_key="airportId",
+            cols=["total_passengers", "company"],
+            operation=["mean", "mode"],
+        )
+    >>> join_agg.fit_transform(main)
+        airportId airportName company_mode  total_passengers_mean
+    0          1   Paris CDG           AF             103.333333
+    1          2      NY JFK           DL              80.000000
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        table: DataFrameLike | Iterable[DataFrameLike] | str | Iterable[str],
+        *,
+        foreign_key: str | Iterable[str],
+        main_key: str | Iterable[str],
+        cols: str | Iterable[str] | None = None,
+        operation: str | Iterable[str] | None = None,
+        suffix: str | Iterable[str] | None = None,
+    ):
+        self.table = table
+        self.foreign_key = foreign_key
+        self.cols = cols
+        self.main_key = main_key
+        self.operation = operation
+        self.suffix = suffix
+
+    def fit(
+        self,
+        X: DataFrameLike,
+        y: ArrayLike | SeriesLike | None = None,
+    ) -> "AggJoiner":
+        """Aggregate auxiliary tables based on the main keys.
+
+        Parameters
+        ----------
+        X : DataframeLike
+            Input data, based table on which to left join the
+            auxiliary tables.
+
+        y : array-like of shape (n_samples), default=None
+            Prediction target. Used to compute correlations between the
+            generated covariates and the target for screening purposes.
+
+        Returns
+        -------
+        AggJoiner
+            Fitted :class:`AggJoiner` instance (self).
+        """
+        self.check_input(X)
+        skrub_px, _ = get_df_namespace(*self.table_)
```

Comment:
`px` is maybe not super obvious and would stop making sense if we add a backend besides pd and pl?

Reply:
Yes, so this is in reference to the array API used in scikit-learn and as a standalone project, naming the namespace variable `xp`, for numpy, cupy and pytorch.

Happy to hear your suggestions if you have some :)

Reply:
as it's the output of `get_df_namespace`, maybe `df_namespace`. but it doesn't really matter as it's just a local variable we can always change it. I just thought it's a name that will come up often so we may pick one that we use all the time to refer to those modules